File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingCalendarNoteColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TrainingCalendarNoteRow } from "./TrainingCalendarNoteRow";

export interface TrainingCalendarNoteColumns {
    TrainingCalendarNoteId: Column<TrainingCalendarNoteRow>;
    Title: Column<TrainingCalendarNoteRow>;
    TrainingCalendarName: Column<TrainingCalendarNoteRow>;
    NoteDate: Column<TrainingCalendarNoteRow>;
    IsActive: Column<TrainingCalendarNoteRow>;
}

export class TrainingCalendarNoteColumns extends ColumnsBase<TrainingCalendarNoteRow> {
    static readonly columnsKey = 'Administration.TrainingCalendarNote';
    static readonly Fields = fieldsProxy<TrainingCalendarNoteColumns>();
}

File: QadeerApp.Web/Modules/Administration/TrainingCalendarNote/TrainingCalendarNoteColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.TrainingCalendarNote")]
[BasedOnRow(typeof(TrainingCalendarNoteRow), CheckNames = true)]
public class TrainingCalendarNoteColumns
{
    [EditLink, AlignRight]
    public int TrainingCalendarNoteId { get; set; }

    [EditLink, Width(220)]
    public string Title { get; set; }

    [Width(220)]
    public string TrainingCalendarName { get; set; }

    [Width(140)]
    public DateTime NoteDate { get; set; }

    [Width(90)]
    public bool IsActive { get; set; }
}
